### PR TITLE
cherrytree: 0.39.4 -> 0.99.37

### LINKED ISSUE #126457: Requesting: CherryTree by Giuspen the latest version 99.37
**Project description**
CherryTree by Giuspen, currently, we are using version 0.39.4, meanwhile the latest application is on 0.99.37. There's a weird jump of version from 0.39.4 to 0.99.13. I don't know why this happens.

**Metadata**

* homepage URL: https://www.giuspen.com/cherrytree/
* source URL: https://github.com/giuspen/cherrytree
* license: GPL3 +
* platforms: Linux

### PR DESCRIPTION
**Pull request by OPNA2608**
###### Motivation for this change
Closes #126457.

Since you seem to use this application, could you test this PR @cicildihar? I've made sure that it lauches on non-NixOS x86_64-linux but I haven't tested anything else about it. I'll give it a shot on x86_64-darwin this evening.

I tried to enable the tests but [this line](https://github.com/giuspen/cherrytree/blob/0.99.37/tests/tests_misc_utils.cpp#L343) gets `application/x-zerosize` instead of the expected `text/plain`. I've reverted the test-related changes for now.

grafik

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 126490` at 08b95eb8 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>cherrytree</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl3` is a deprecated license, please check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/applications/misc/cherrytree/default.nix:64:5:
```
   |
64 |     license = licenses.gpl3;
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/applications/misc/cherrytree/default.nix:65:5:
```
   |
65 |     maintainers = with maintainers; [ ];
   |     ^
```
</li>
  </ul>
</details>

File: pkgs/applications/misc/cherrytree/default.nix [hunk 1]
```diff
@@ -31,7 +59,8 @@ python2Packages.buildPythonApplication rec {
       you have scattered around your hard drive can be conveniently placed into
       a Cherrytree document where you can easily find it.
     '';
-    homepage = "http://www.giuspen.com/cherrytree";
+    homepage = "https://www.giuspen.com/cherrytree";
+    changelog = "https://raw.githubusercontent.com/giuspen/cherrytree/${version}/changelog.txt";
     license = licenses.gpl3;
```
Comment (fabaff) [inline pkgs/applications/misc/cherrytree/default.nix:64]: The license is `GPLv3+` according to the [file headers](https://github.com/giuspen/cherrytree/blob/master/src/ct/ct_actions.h#L8-L11).

```suggestion
    license = licenses.gpl3Plus;
```

**Comment by cicildihar** [comment]
> could you test this PR @cicildihar?

yes would be happy to do that. it's just that I don't know how to. I've tried it with nixpkgs-review pr 126490 and nix-shell -p but all that it says that i need to be in nixpkgs directory. So, should I clone nixpkgs first then?

**Comment by cicildihar** [comment]
Update: I've tested this PR with my NixOS based machine and it works fine against documents from version 99.x.

**Comment by OPNA2608** [comment]
As-is I don't think I can backport it. I tried building on Darwin and it failed: gtksourceviewmm andspeak are marked Linux-only, gspell doesn't compile if I enable it and that blocks cherrytree from getting past the cmake phase. For Darwin users this would be a bump that removes this package.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 126490` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>cherrytree</li>
  </ul>
</details>

**Comment by OPNA2608** [comment]
> I'll give it another shot after work.

Sorry, I was too busy working on other things and forgot about it. I got it working on Darwin now, will push those commits in abit.

Screenshot_macOS_2021-06-13_21:49:58

File: pkgs/development/libraries/gspell/default.nix [hunk 2]
```diff
@@ -23,18 +26,24 @@ stdenv.mkDerivation rec {
...
+    ./0001-Darwin-build-fix.patch
+  ];
+
   nativeBuildInputs = [
     pkg-config
     vala
     gobject-introspection
     libxml2
+    autoreconfHook
+    gnome2.gtk-doc
```
Comment (jtojnar) [inline pkgs/development/libraries/gspell/default.nix:39]:
```suggestion
    gtk-doc
```

File: pkgs/development/libraries/gspell/default.nix [hunk 3]
```diff
@@ -23,18 +26,24 @@ stdenv.mkDerivation rec {
...
   ];
 
   buildInputs = [
     glib
     gtk3
     icu
-  ];
+  ] ++ lib.optional stdenv.hostPlatform.isDarwin gtk-mac-integration;
```
Comment (jtojnar) [inline pkgs/development/libraries/gspell/default.nix:46]:
```suggestion
  ] ++ lib.optionals stdenv.hostPlatform.isDarwin [
    gtk-mac-integration
  ];
```

File: pkgs/development/libraries/gspell/default.nix [hunk 4]
```diff
@@ -23,18 +26,24 @@ stdenv.mkDerivation rec {
     sha256 = "1pdb4gbjrs8mk6r0ipw5vxyvzav1wvkjq46kiq53r3nyznfpdfyw";
   };
 
+  patches = [
+    ./0001-Darwin-build-fix.patch
```
Comment (jtojnar) [inline pkgs/development/libraries/gspell/default.nix:30]: Could you please try submitting the patch upstream?

Comment (OPNA2608) [reply pkgs/development/libraries/gspell/default.nix:30]: https://gitlab.gnome.org/GNOME/gspell/-/blob/30f66b828173c6ea7c1f413503d80e8fb02cc74e/HACKING

```
gspell development
==================

Contributions are no longer accepted
------------------------------------

And new maintainers are not accepted either.
```

<details>
  <summary>I was also trying to get the latest gspell commit to build so I could make sure the patch works there too before I submit it, but it just spits autoconf errors at me that I can't solve.</summary>

```
autoreconf: running: /nix/store/1hbdwchm6mkqhn8ygdgjjj9kkmps8ilv-autoconf-2.71/bin/autoconf --force
configure.ac:58: warning: The macro `AC_CONFIG_HEADER' is obsolete.
configure.ac:58: You should run autoupdate.
./lib/autoconf/status.m4:719: AC_CONFIG_HEADER is expanded from...
configure.ac:58: the top level
configure.ac:126: warning: AX_ADD_AM_MACRO_STATIC is m4_require'd but not m4_defun'd
m4/ax_code_coverage.m4:208: _AX_CODE_COVERAGE_ENABLED is expanded from...
lib/m4sugar/m4sh.m4:699: AS_IF is expanded from...
m4/ax_code_coverage.m4:248: AX_CODE_COVERAGE is expanded from...
configure.ac:126: the top level
configure.ac:150: warning: The macro `AC_TRY_COMPILE' is obsolete.
configure.ac:150: You should run autoupdate.
./lib/autoconf/general.m4:2847: AC_TRY_COMPILE is expanded from...
configure.ac:150: the top level
configure.ac:70: error: possibly undefined macro: PKG_PROG_PKG_CONFIG
      If this token and others are legitimate, please use m4_pattern_allow.
      See the Autoconf documentation.
configure.ac:71: error: possibly undefined macro: PKG_INSTALLDIR
configure:16724: error: possibly undefined macro: AC_MSG_ERROR
configure:17035: error: possibly undefined macro: AM_V_at
configure:17053: error: possibly undefined macro: AM_DISTCHECK_CONFIGURE_FLAGS
autoreconf: error: /nix/store/1hbdwchm6mkqhn8ygdgjjj9kkmps8ilv-autoconf-2.71/bin/autoconf failed with exit status: 1
```

</details>

Comment (OPNA2608) [reply pkgs/development/libraries/gspell/default.nix:30]: The original commit i looked at that fixed gspell (https://github.com/mjlbach/nixpkgs/commit/5cf11fc91af7effed5cb2ccb3d4cb9ffa0e100b3) didn't specify where the patch is from but it [seems to be from Homebrew, embedded in their build script](https://github.com/Homebrew/homebrew-core/blob/2a27fb86b08afc7ae6dff79cf64aafb8ecc93275/Formula/gspell.rb#L125-L149). I can try to fetchpatch their recipe & extract the patch in postFetch instead if you'd prefer that?

Comment (SuperSandro2000) [reply pkgs/development/libraries/gspell/default.nix:30]: At least add a link from where you got the patch incase it ever needs an update.

Comment (OPNA2608) [reply pkgs/development/libraries/gspell/default.nix:31]: Done.

**Comment by OPNA2608** [comment]
Gonna backport this since the version on stable is a year old, the changes to gspell and gtksourceviewmm mean that there won't be a regression in support for Darwin users and the big difference in version seems to be due to the port to a different language and not incompatibilities.

**Pull request merged**